### ISSUE TITLE
Feature request: add support for nodejs16.x runtime

### ISSUE THREAD

**Issue by saragerion** [aws-powertools/powertools-lambda-typescript#830]
## Description of the feature request

**Problem statement**

Support for nodejs16.x currently missing.

**Summary of the feature**

The runtime nodejs16.x will be GA this week, see this [related issue](https://github.com/aws/aws-lambda-base-images/issues/14).

As part of this issue, support for the Node.js v16 runtime should be added where relevant. 

**Code examples**

N/A

**Benefits for you and the wider AWS community**

N/A

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

### Related issues, RFCs

https://github.com/aws/aws-lambda-base-images/issues/14

**Comment by dreamorosi** [comment]
From a first round of research, here are some places that needs updating:
- [`.devcontainer/Dockerfile`](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.devcontainer/Dockerfile)
- [`.devcontainer/devcontainer.json`](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.devcontainer/devcontainer.json)
- [`.github/workflows/on-merge-to-main.yml`](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/workflows/on-merge-to-main.yml#L24-L27)
- [`.github/workflows/on-release-prod.yml`](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/workflows/on-release-prod.yml#L21-L24)
- [`.github/workflows/pr_lint_and_test.yml`](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/workflows/pr_lint_and_test.yml#L14-L15)
- [`.github/workflows/run-e2e-tests.yml`](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/workflows/run-e2e-tests.yml)
- [`examples/cdk/src/example-function.ts`](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/examples/cdk/src/example-function.ts)
- [`examples/sam/template.yaml`](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/examples/sam/template.yaml#L18)
- [`CONTRIBUTING.md`](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/CONTRIBUTING.md)

**Comment by dreamorosi** [comment]
At the moment we are blocked by CDK releasing support for the new runtime (we use CDK in the integration tests). 
The PR that adds support has been already merged yesterday https://github.com/aws/aws-cdk/pull/20261 so hopefully it'll be released soon.

**Comment by willfarrell** [comment]
For those that run across this that use terraform: https://github.com/hashicorp/terraform-provider-aws/issues/24714

**Comment by saragerion** [comment]
On hold (hopefully for a relatively short time) until the CDK release containing this update: https://github.com/aws/aws-cdk/pull/20261

**Comment by dreamorosi** [comment]
Both CDK and SAM added support for the new runtime, will link a PR to the issue shortly.
